Jenkinsfile

```diff
@@ -78,6 +86,24 @@ pipeline {
                 }
             }
         }
+
+        stage('Upload APK') {
+            when {
+                expression { !params.ghprbPullId && params.UPLOAD_APK }
+            }
+
+            steps {
+	        script {
+                    echo "Upload the built Kolibri Android packages for internal testers"
+                    build(
+                        job: 'kolibri-installer-android-upload',
+                        parameters: [
+                            string(name: 'RELEASE_NOTES', value: '[]'),
+                        ]
+                    )
```

Comment:
I don't think this will actually work correctly as the upload job depends on the saved artifact from the last successful job. Since this job hasn't completed, it would likely use the artifacts from the previously successful job.

Instead, I think you need to move this down to the `post` block within a [`success`](https://www.jenkins.io/doc/book/pipeline/syntax/#post-conditions) condition block. However, you likely need to repeat the same nonsense as the failure block to wrap it in a `script` block and test the conditional. I think it would end up looking like this:

```
post {
    success {
        script {
            if (!params.ghprbPullId && params.UPLOAD) {
                echo "Upload the built Kolibri Android packages for internal testers"
                build(
                    job: 'kolibri-installer-android-upload',
                    parameters: [
                        string(name: 'RELEASE_NOTES', value: '[]'),
                    ]
                )
            }
        }
    }

    failure {
        ...
    }
}
```

Reply:
make sense!